OpenRA.Mods.Common/Scripting/Global/DateTimeGlobal.cs

```diff
@@ -32,16 +38,26 @@ public int GameTime
 			get { return Context.World.WorldTick; }
 		}
 
-		[Desc("Converts the number of seconds into game time (ticks).")]
-		public int Seconds(int seconds)
+		[Desc("Converts the number of seconds into game time (ticks), measured in time passed in-game by default,",
+			"or by real time if `walltime` is true.")]
+		public int Seconds(int seconds, bool walltime = false)
+		{
+			return seconds * (walltime ? 1000 / Context.World.Timestep : 25);
+		}
+
+		[Desc("Converts the number of minutes into game time (ticks), measured in time passed in-game by default,",
+			"or by real time if `walltime` is true.")]
+		public int Minutes(int minutes, bool walltime = false)
 		{
-			return seconds * 25;
+			return Seconds(minutes * 60, walltime);
 		}
 
-		[Desc("Converts the number of minutes into game time (ticks).")]
-		public int Minutes(int minutes)
+		[Desc("Set the game's time limit. Add `GameTime.GameTicks` to the value when setting the limit mid-game",
```

Comment:
Having to manually add the current game ticks feels dirty, IMO. When called mid-game I expect that the most common usecase will be to start the timer from then, so if anything we should be asking users to subtract the GameTicks if they really did want it relative to the start.

IMO it would be better to have it explicit, with a `SetTimeLimit(int ticks, bool relative=true)` method instead of a property setter

Reply:
done